airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -633,36 +670,39 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
 
 
 class BulkIncrementalSalesforceStream(BulkSalesforceStream, IncrementalRestSalesforceStream):
+
     def next_page_token(self, last_record: Mapping[str, Any]) -> Optional[Mapping[str, Any]]:
         if self.name not in UNSUPPORTED_FILTERING_STREAMS:
-            page_token: str = last_record[self.cursor_field]
-            res = {"next_token": page_token}
-            # use primary key as additional filtering param, if cursor_field is not increased from previous page
-            if self.primary_key and self.prev_start_date == page_token:
-                res["primary_key"] = last_record[self.primary_key]
-            return res
+            return {
+                "next_token": last_record[self.cursor_field],
+                "primary_key": last_record.get(self.primary_key)
+            }
         return None
 
     def request_params(
         self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
     ) -> MutableMapping[str, Any]:
-        selected_properties = self.get_json_schema().get("properties", {})
+        start_date = max(
+            (stream_state or {}).get(self.cursor_field, ""),
+            (stream_slice or {}).get("start_date", ""),
+            (next_page_token or {}).get("start_date", "")
+        )
+        end_date = stream_slice["end_date"]
 
-        stream_date = stream_state.get(self.cursor_field)
-        next_token = (next_page_token or {}).get("next_token")
-        primary_key = (next_page_token or {}).get("primary_key")
-        start_date = next_token or stream_date or self.start_date
-        self.prev_start_date = start_date
+        select_fields = ", ".join(self.get_json_schema().get('properties', {}).keys())
+        table_name = self.name
+        where_conditions = [f"{self.cursor_field} >= {start_date}", f"{self.cursor_field} < {end_date}"]
+        order_by_clause = ""
 
-        query = f"SELECT {','.join(selected_properties.keys())} FROM {self.name} "
-        if start_date:
-            if primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
-                query += f"WHERE ({self.cursor_field} = {start_date} AND {self.primary_key} > '{primary_key}') OR ({self.cursor_field} > {start_date}) "
-            else:
-                query += f"WHERE {self.cursor_field} >= {start_date} "
         if self.name not in UNSUPPORTED_FILTERING_STREAMS:
-            order_by_fields = [self.cursor_field, self.primary_key] if self.primary_key else [self.cursor_field]
-            query += f"ORDER BY {','.join(order_by_fields)} ASC LIMIT {self.page_size}"
+            primary_key = (next_page_token or {}).get("primary_key", "")
```

Comment:
nit: can you rename this variable to `last_key` for clarity

Reply:
done